src/inference/enumerate.ad.js

```diff
@@ -47,6 +57,17 @@ module.exports = function(env) {
     return this.wpplFn(_.clone(this.store), env.exit, this.a);
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  Enumerate.prototype.error = function(errType) {
```

Comment:
If you don't want to return strings when inference functions fail, you could also do something like this:

```js
var err = new Error(errType);
if (this.throwOnError) {
  throw err;
} else {
  return this.k(this.store, err);
}
```

Then you can check `returnVal instanceof Error` to determine whether the inference function failed.